### PR TITLE
Admin User

### PR DESCRIPTION
**Pull request by redwarewolf**
## Summary

- Created Admin Users

### CODE REVIEW

File: app/routes.js [hunk 1]
```diff
@@ -14,6 +16,16 @@ exports.init = app => {
...
     usersController.create
   );
+  app.post(
+    `${URL}/admin/users`,
+    [
+      paramsValidator.validateSchemaAndFail(schemas.admins.create),
+      adminValidator.create,
+      adminValidator.validateAdmin,
+      authorizationValidator.validate
+    ],
```
Comment (picollomartin) [inline app/routes.js:26]: I think you need only two middlewares, one for check the schema and another for the token and the permission necessary

File: app/middlewares/validators/adminValidator.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  jwt.findUserByToken(req.headers.authorization).then(user => {
+    if (user.type !== 'admin') {
+      next(errors.authenticationError('User is not an Admin'));
+    }
+    next();
+  });
+};
+
+exports.create = (req, res, next) => {
+  User.findByPk(req.body.user_id).then(user => {
```
Comment (picollomartin) [inline app/middlewares/validators/adminValidator.js:15]: Be careful with the databases errors handling, you're not caching nothing (this applies to several parts in this same PR)

File: app/middlewares/userValidator.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+const { User } = require('../models/index');
+const errors = require('../errors');
+
+exports.validate = (req, res, next) => {
```
Comment (picollomartin) [inline app/middlewares/userValidator.js:4]: I don't know why do you need this middleware

File: test/controllers/admin.test.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+              .send({ user_id: user.id })
+              .set({
+                Accept: 'application/json',
+                authorization:
+                  'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJtYWlsIjoicGVkcm8uamFyYUB3b2xveC5jb20uYXIifQ.zLLy2i25xQZuXyk0s98afCQA4hlomRq92D1lZQcP-mE'
+              })
+              .then(response => {
+                expect(response.status).toBe(200);
+                expect(response.body.id).toBe(user.id);
+                done();
```
Comment (picollomartin) [inline test/controllers/admin.test.js:27]: Check the admin type in the database

File: test/controllers/admin.test.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+          .then(res => {
+            request
+              .post('/api/v1/admin/users')
+              .send({ user_id: user.id })
+              .set({
+                Accept: 'application/json',
+                authorization: res.headers.authorization
+              })
+              .then(response => {
+                expect(response.status).toBe(500);
```
Comment (picollomartin) [inline test/controllers/admin.test.js:88]: 5xx ? Seems like a forbidden error or something like that

File: app/middlewares/validators/userValidator.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+const { User } = require('../../models/index');
+const errors = require('../../errors');
+
+exports.validate = (req, res, next) => {
+  User.findByPk(req.headers.user_id).then(user => {
```
[the remark this answers is not in the document]
Comment (redwarewolf) [reply app/middlewares/validators/userValidator.js:5]: Yes! My bad hehe

File: app/middlewares/validators/authorizationValidator.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
+const jwt = require('../../services/jwt');
+const errors = require('../../errors');
+
+exports.validate = (req, res, next) => {
```
[the remark this answers is not in the document]
Comment (redwarewolf) [reply app/middlewares/validators/authorizationValidator.js:4]: I did include this to validate authorizations and admin tokens

File: app/middlewares/userValidator.js (hunk 3 above)
Comment (redwarewolf) [reply app/middlewares/userValidator.js:4]: Forgot to remove it

File: app/services/users.js [hunk 8]
```diff
@@ -28,3 +28,5 @@ exports.createSession = async params => {
   logger.error('Password and mail mismatch for user:', params.mail);
   throw errors.authenticationError("The password and mail combination doesn't match");
 };
+
+exports.findUserByMail = mail => User.findOne({ where: { mail } });
```
Comment (picollomartin) [inline app/services/users.js:32]: Add a catch here

**Pull request merged**